File: src/types/appConfig.ts
import type { RootFolder } from './rootFolder'
import type { EditorTheme } from './editor'
import type { CustomEditorLayout, LayoutDensity } from './settings'

export type LastWorkspaceState = {
  rootFolderPath?: string
  selectedFolderPath?: string
  selectedFilePath?: string
  activeSystemFolder?: 'all-files' | 'recently-deleted' | null
  leftSidebarScrollTop?: number
  middleFileListScrollTop?: number
  rightEditorScrollTop?: number
  rightEditorScrollHeight?: number
  rightEditorClientHeight?: number
  sidebarCollapsed?: boolean
  leftPanelWidth?: number
  middlePanelWidth?: number
  sortMode?: string
  editorMode?: 'readonly' | 'editing'
  updatedAt: string
}

export type AppConfig = {
  rootFolders: RootFolder[]
  selectedRootFolderId?: string
  selectedFolderPath?: string
  selectedFilePath?: string
  pinnedFilePaths?: string[]
  layoutDensity?: LayoutDensity
  customEditorLayout?: CustomEditorLayout
  editorTheme?: EditorTheme
  sidebarCollapsed?: boolean
  lastWorkspaceState?: LastWorkspaceState
}

File: src-tauri/src/models/app_config.rs
use serde::{Deserialize, Serialize};

use super::root_folder::RootFolderPayload;

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct AppConfig {
    pub root_folders: Vec<RootFolderPayload>,
    pub selected_root_folder_id: Option<String>,
    pub selected_folder_path: Option<String>,
    pub selected_file_path: Option<String>,
    pub layout_density: Option<String>,
    pub editor_theme: Option<String>,
    pub sidebar_collapsed: Option<bool>,
}

